projects/dsp-ui/src/lib/action/components/select-project/select-project.component.ts

```diff
@@ -0,0 +1,124 @@
+import { Component, EventEmitter, Inject, Input, OnInit, Output } from '@angular/core';
+import { FormBuilder, FormControl, FormGroup } from '@angular/forms';
+import { ApiResponseData, ApiResponseError, KnoraApiConnection, ProjectsResponse, ReadProject } from '@dasch-swiss/dsp-js';
+import { Observable } from 'rxjs';
+import { map, startWith } from 'rxjs/operators';
+import { DspApiConnectionToken } from '../../../core';
+import { SortingService } from '../../services/sorting.service';
+
+@Component({
+    selector: 'dsp-select-project',
+    templateUrl: './select-project.component.html',
+    styleUrls: ['./select-project.component.scss']
+})
+export class SelectProjectComponent implements OnInit {
+
+    // user IRI so we could show only projects that belongs to this user
+    @Input() userIri?: string;
+
+    // selected project
+    @Output() selectedProject: EventEmitter<ReadProject> = new EventEmitter<ReadProject>();
+
+    // in case of an (api) error
+    error: any;
+
+    // filter projects while typing (autocomplete)
+    filteredProjects: Observable<ReadProject[]>;
+
+    // list of projects
+    projects: ReadProject[];
+
+    // form group
+    selectProjectForm: FormGroup;
+
+    constructor(
+        @Inject(DspApiConnectionToken) private _dspApiConnection: KnoraApiConnection,
+        private _sortingService: SortingService,
+        private _formBuilder: FormBuilder
+    ) {}
+
+    ngOnInit(): void {
+
+        // clean autocomplete list
+        this.projects = [];
+
+        // get list of all projects
+        this.getAllProjects();
+
+        this.selectProjectForm = this._formBuilder.group({
+            projectName: new FormControl({
+                value: '', disabled: false
+            }) as FormControl
```

Comment:
What do you need the [type assertion](https://www.typescriptlang.org/docs/handbook/basic-types.html#type-assertions) `FormControl` for? Did the compiler complain? 

Is the `disabled: false` needed or could you just do `new FormControl('')` it the empty string is your initial value?

Reply:
> Instantiate a FormControl, with an initial value.
> const control = new FormControl('some value');
> console.log(control.value);     // 'some value'

https://angular.io/api/forms/FormControl#initializing-form-controls